ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -386,8 +394,11 @@ defRowToSymbolInfo (DefRow{..}:.(modInfoSrcFile -> Just srcFile))
     end   = Position (defELine - 1) (defECol - 1)
 defRowToSymbolInfo _ = Nothing
 
-pointCommand :: HieASTs t -> Position -> (HieAST t -> a) -> [a]
-pointCommand hf pos k =
+pointCommand :: HieASTs t -- ^ Get AST
+  -> Position -- ^ point at position
+  -> (HieAST t -> a) -- ^ extact data with a getter
+  -> [a] -- ^ all data recieved
```

Comment:
I don't think these comments are very informative.

Reply:
But for me they are.

The lack of information (or superficiality of information) in the documentation for the reader - means that the next reader would find new information & would add it into documentation.

I am just trying to doc functions so I understand them & later can work with them.

I needed to go look into the sources of all these arguments. Sometimes I can do just jump to definition, sometimes I need to go Hoogle & frequently Hoogle returns a GHC 9.2.1 interface & I develop in 8.10.7, so I need to go find the definitions/sources manually to look into them.

I am a newb in the project.

And the docs are for newbs. When person implicitly knows what the function does, or they understand all they need to know from the type signature - on that level of understanding docs are not needed & so the person does not look into docs.

So if left untouched the docs tend to go from nothing there to something there, converging on the medium level, since above medium level understanding no longer needs to look-up medium level info or add to it.

For me - it is useful information. `point` helps to indicate that `Position` is a main topic of the function, that `(HieAST t -> a)` is a getter that extracts info from `AST` based on `Position` & returns plural.

& noting that `point` is `Position` also helps to understand what function is about.

Because signature can have different implementations. It can take AST & `(HieAST t -> a)` is a pointer to something & `Position` is a filter identifier & then the body of function is a `Command`.

Doc explains that `pointCommand` is a command when user points at some position - information is retried through getter.

The question generally is not to remove documentation, but how to modify it so it optimally explains the function. Before this, I did not understand how `pointCommand` & `(HieAST t -> a)` combine.

Reply:
> The question generally is not to remove documentation, but how to modify it so it optimally explains the function.

If they are not informative, I'm more inclined to *remove* documentations. They give readers extra clutters to read. Especially that `getter` part. It can compute anything from `HieAST`. It is in a sense "getter", but it explains nothing more than the actual type. The type already says that the argument is the function which accepts AST and computes something.

One issue of these comments, IMO, is that the target of explanation is unclear. For example, `point at position` comment. This one does not describe `Position` argument, but some `point` (which you probably meant to refer to `point` of `pointCommand`).
    - If the description is for the position argument, please describe it. For example, `Position of the target point`, or `Position where the command will be executed`, etc. Although I don't think even these are good comments, they at least explain the position argument, not "point".
    - If the description is for the entire function, why do you put those comments on the arguments? You can put a document for the function itself.

Reply:
Actually I'm not sure what you mean by
> Because signature can have different implementations. It can take AST & (HieAST t -> a) is a pointer to something & Position is a filter identifier & then the body of function is a Command, or other readings of the signature.

Your comments are not any better than the types. If `HieAST t -> a` can be a pointer, than that "getter" can also be a pointer. If `Position` is a filter identifier, then what's the difference of `position of point` (I swapped the words so that it actually explains the argument)?

Reply:
That is not a final form of documentation, documentation can be easily modified.

With documentation, both sides need to propose improvements. So far received only "remove" suggestions, which is not helpful to improve documentation, if to have an inverse Socratic reflection.

Because I get not suggestions of how to improve the situation - I need to improve it myself, which results in new versions of the documentation, while old thread addressed the previous version.

Reply:
> which is not helpful to improve documentation

As I said, putting uninformative documentation goes against *code quality*. Surely, it does not improve *documentation*, but it at least protects *code quality*.

> Because I get not suggestions of how to improve the situation

I cannot suggest you a comment that helps you to understand, as I am not you. What I can do is just arguing against bad and possibly unnecessary comments.

Let me summarize my comments.

1. If you want to explain a function, please use a function doc comment.
2. If you want to explain an argument, please explain the argument, not its type.
3. If you want to explain an argument type, please put that on the type definition, not on the usage.
4. Even in any of the above cases, please do not just repeat what already being said in a type or a function name. For example, in the case of `Position`, one possibility is that mentioning that `Position` is actually where we find a smallest containing node to which we apply those "getter"s.

Reply:
Ok. Deleted the docs. Done.